### PR TITLE
feat: commandOptionsParser util

### PR DESCRIPTION
**Pull request by Skillz4Killz**

### CODE REVIEW

**Comment by Skillz4Killz** [comment]
@H01001000 Hey can you look at the issue here in deno unit test plz

File: packages/utils/tests/interactions.spec.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
+import { ApplicationCommandOptionTypes, ApplicationCommandTypes, ButtonStyles, InteractionTypes, type DiscordInteraction } from '@discordeno/types'
+import { expect } from 'chai'
+import { describe } from 'mocha'
```
Comment (H01001000) [inline packages/utils/tests/interactions.spec.ts:3]:
```suggestion
import { describe, it } from 'mocha'
```

**Review by H01001000 — CHANGES_REQUESTED** [review]

**Review by H01001000 — DISMISSED** [review]

File: packages/utils/src/interactions.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { ApplicationCommandOptionTypes, type DiscordInteraction } from '@discordeno/types'
+
+export function commandOptionsParser(interaction: DiscordInteraction, options?: NonNullable<DiscordInteraction['data']>['options']): Record<string, any> {
+  if (!interaction.data) return {}
+  if (!options) options = interaction.data.options ?? []
+
+  const args: Record<string, any> = {}
+
+  for (const option of options) {
+    switch (option.type) {
```
Comment (itohatweb) [inline packages/utils/src/interactions.ts:10]: Please add support and tests for those types too:
```ts
Boolean,
Mentionable,
Number,
Attachment,
```

**Review by itohatweb — CHANGES_REQUESTED** [review]

**Comment by Skillz4Killz** [comment]
@itohatweb ready to merge

**Review by itohatweb — APPROVED** [review]

**Pull request merged**